azurerm/resource_arm_cosmosdb_mongo_collection.go

```diff
@@ -243,11 +258,14 @@ func resourceArmCosmosDbMongoCollectionRead(d *schema.ResourceData, meta interfa
 
 	throughputResp, err := client.GetMongoDBCollectionThroughput(ctx, id.ResourceGroup, id.Account, id.Database, id.Collection)
 	if err != nil {
-		return fmt.Errorf("Error reading Throughput on Cosmos Mongo Collection %s (Account %s, Database %s): %+v", id.Collection, id.Account, id.Database, err)
-	}
-
-	if throughput := throughputResp.Throughput; throughput != nil {
-		d.Set("throughput", int(*throughput))
+		if !utils.ResponseWasNotFound(throughputResp.Response) {
+			_ = d.Set("throughput", nil)
+			return fmt.Errorf("Error reading Throughput on Cosmos Mongo Collection %s (Account %s, Database %s): %+v", id.Collection, id.Account, id.Database, err)
+		}
+	} else {
+		if throughput := throughputResp.Throughput; throughput != nil {
+			_ = d.Set("throughput", int(*throughput))
```

Comment:
d.Set() will deal with nil values so:
```suggestion
			d.Set("throughput", throughputResp.Throughput)
```